tests/unit/test_bruteforce_multi.cpp

```diff
@@ -179,251 +179,296 @@ TEST_F(BruteForceMultiTest, brute_force_empty_index) {
+    }
+    ASSERT_EQ(VecSimIndex_IndexSize(index), n);
 
-//     for (size_t i = 0; i < n; i++) {
-//         float f[dim];
-//         for (size_t j = 0; j < dim; j++) {
-//             f[j] = (float)(i / 10); // i / 10 is in integer (take the "floor" value).
-//         }
-//         VecSimIndex_AddVector(index, (const void *)f, i);
-//     }
-//     ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+    float query[] = {0, 0, 0, 0};
+    auto verify_res = [&](size_t id, float score, size_t index) { ASSERT_EQ(id, index); };
+    runTopKSearchTest(index, query, k, verify_res);
+    auto res = VecSimIndex_TopKQuery(index, query, k, nullptr, BY_SCORE);
+    auto it = VecSimQueryResult_List_GetIterator(res);
+    for (size_t i = 0; i < k; i++) {
+        auto el = VecSimQueryResult_IteratorNext(it);
+        labelType element_label = VecSimQueryResult_GetId(el);
+        ASSERT_EQ(element_label, i);
+        idType element_id =
+            reinterpret_cast<BruteForceIndex_Multi *>(index)->labelToIdsLookup.at(element_label)[0];
+        ASSERT_EQ(element_id, i * perLabel);
+    }
+    VecSimQueryResult_IteratorFree(it);
+    VecSimQueryResult_Free(res);
+    VecSimIndex_Free(index);
+}
 
-//     // Run a query where all the results are supposed to be {5,5,5,5} (different ids).
-//     float query[] = {4.9, 4.95, 5.05, 5.1};
-//     auto verify_res = [&](size_t id, float score, size_t index) {
-//         ASSERT_TRUE(id >= 50 && id < 60 && score <= 1);
-//     };
-//     runTopKSearchTest(index, query, k, verify_res);
+TEST_F(BruteForceMultiTest, find_better_score) {
+    size_t n = 100;
+    size_t k = 10;
+    size_t n_labels = 10;
+    size_t dim = 4;
+    size_t initial_capacity = 200;
 
-//     VecSimIndex_Free(index);
-// }
+    VecSimParams params{.algo = VecSimAlgo_BF,
+                        .bfParams = BFParams{.type = VecSimType_FLOAT32,
+                                             .dim = dim,
+                                             .metric = VecSimMetric_L2,
+                                             .multi = true,
+                                             .initialCapacity = initial_capacity}};
+    VecSimIndex *index = VecSimIndex_New(&params);
 
-// TEST_F(BruteForceMultiTest, brute_force_reindexing_same_vector) {
-//     size_t n = 100;
-//     size_t k = 10;
-//     size_t dim = 4;
-//     size_t initial_capacity = 200;
+    // Building the index. Each label gets 10 vectors with decreasing (by insertion order) element
+    // value, so when we search, each vector is better then the previous one. Furthermore, each
+    // label gets at leat one better vector then the previous label, and one with score equals to
+    // the best of the previous label, so the multimap holds at least two labels with the same
+    // score.
+    std::map<size_t, float> scores;
+    for (size_t i = 0; i < n; i++) {
+        float f[dim];
+        size_t el = ((n - i - 1) % n_labels) + ((n - i) / n_labels);
```

Comment:
Can you explain (in a comment) how for example, label 1 gets one vector with a score equal to label 2?